conversations.go

```diff
@@ -0,0 +1,175 @@
+		res := conv.Title == in.Title
+		if res {
+			return nil, status.Error(codes.Internal, "a conversations with this title always exist")
+		}
+	}
+
+	conversation, err := server.repo.Create(ctx, &models.ConversationInfo{Title: in.Title, GroupID: in.GroupId})
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+	return &conversationsv1.CreateConversationResponse{
+		Conversation: &conversationsv1.Conversation{
+			Id:      conversation.ID,
+			GroupId: conversation.GroupID,
+			Title:   conversation.Title,
+		},
+	}, nil
+}
+
+func (server *conversationsAPI) GetConversation(ctx context.Context, in *conversationsv1.GetConversationRequest) (*conversationsv1.GetConversationResponse, error) {
+	err := validators.ValidateGetConversationRequest(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "could not get Conversation")
+	}
+
+	conversation, err := server.repo.Get(ctx, &models.OneConversationFilter{ID: in.ConversationId})
+	if err != nil {
+		server.logger.Error("failed to get conversation from conversation id", zap.Error(err))
+	}
+
+	return &conversationsv1.GetConversationResponse{
+		Conversation: &conversationsv1.Conversation{
+			Id:      conversation.ID,
+			GroupId: conversation.GroupID,
+			Title:   conversation.Title,
+		},
+	}, nil
+}
+
+func (server *conversationsAPI) DeleteConversation(ctx context.Context, in *conversationsv1.DeleteConversationRequest) (*conversationsv1.DeleteConversationResponse, error) {
+	err := validators.ValidateDeleteConversationRequest(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	err = server.repo.Delete(ctx, &models.OneConversationFilter{ID: in.ConversationId})
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+
+	return &conversationsv1.DeleteConversationResponse{}, nil
+}
+
+func (server *conversationsAPI) UpdateConversation(ctx context.Context, in *conversationsv1.UpdateConversationRequest) (*conversationsv1.UpdateConversationResponse, error) {
+	err := validators.ValidateUpdateConversationRequest(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "could not validate update conversation request")
+	}
+
+	filter := models.OneConversationFilter{ID: in.ConversationId}
+
+	conversation, err := server.repo.Get(ctx, &filter)
+	if err != nil {
+		return nil, status.Error(codes.Internal, "failed to get conversation to update")
+	}
+
+	if conversation.Title == in.Title {
+		return nil, status.Error(codes.Internal, "new title same as old title")
+	}
+
+	groupConversations, err := server.repo.List(ctx, &models.AllConversationsFilter{ID: conversation.GroupID})
+	if err != nil {
+		return nil, status.Error(codes.Internal, "failed to get list of all conversations of group")
+	}
+
+	for _, conv := range groupConversations {
+		res := conv.Title == in.Title
+		if res {
+			return nil, status.Error(codes.Internal, "a conversations with this title always exist")
+		}
+	}
```

Comment:
Not important. Two conversations can have the same name for now. Besides, if in the future we want to prevent two conversations from sharing the same name we can do so in MongoDB directly by imposing a uniqueness constraint on the (group_id, title) attribute pair.

Reply:
Okay didn't know that we could do that in mongo